monkey/monkey_island/cc/resources/island_mode.py

```diff
@@ -21,8 +21,7 @@ def __init__(self, island_mode_service: IslandModeService):
     @jwt_required
     def put(self):
         try:
-            body = json.loads(request.data)
-            mode = IslandModeEnum(body.get("mode"))
+            mode = IslandModeEnum(json.loads(request.data))
```

Comment:
What about:

```suggestion
            mode = IslandModeEnum(request.json)
```

Reply:
Won't work, this is what the request looks like:
```
{'_cached_data': b'"unset"',
 'cookies': ImmutableMultiDict([]),
 'data': b'"unset"',
 'environ': {'CONTENT_LENGTH': '7',
             'HTTP_CONTENT_LENGTH': '7',
             'HTTP_HOST': 'localhost',
             'HTTP_USER_AGENT': 'werkzeug/2.0.2',
             'PATH_INFO': '/api/island/mode',
             'QUERY_STRING': '',
             'RAW_URI': '/api/island/mode',
             'REMOTE_ADDR': '127.0.0.1',
             'REQUEST_METHOD': 'PUT',
             'REQUEST_URI': '/api/island/mode',
             'SCRIPT_NAME': '',
             'SERVER_NAME': 'localhost',
             'SERVER_PORT': '80',
             'SERVER_PROTOCOL': 'HTTP/1.1',
             'flask._preserve_context': True,
             'werkzeug.request': <Request 'http://localhost/api/island/mode' [PUT]>,
             'wsgi.errors': <_io.TextIOWrapper encoding='UTF-8'>,
             'wsgi.input': <_io.BytesIO object at 0x0000019DACA3C828>,
             'wsgi.multiprocess': False,
             'wsgi.multithread': False,
             'wsgi.run_once': False,
             'wsgi.url_scheme': 'http',
             'wsgi.version': (1, 0)},
 'files': ImmutableMultiDict([]),
 'form': ImmutableMultiDict([]),
 'headers': EnvironHeaders([('User-Agent', 'werkzeug/2.0.2'), ('Host', 'localhost'), ('Content-Length', '7')]),
 'host': 'localhost',
 'method': 'PUT',
 'path': '/api/island/mode',
 'query_string': b'',
 'remote_addr': '127.0.0.1',
 'root_path': '',
 'scheme': 'http',
 'server': ('localhost', 80),
 'shallow': False,
 'stream': <werkzeug.wsgi.LimitedStream object at 0x0000019DACAAEFC8>,
 'url': 'http://localhost/api/island/mode',
 'url_rule': <Rule '/api/island/mode' (OPTIONS, PUT, GET, HEAD) -> islandmode>,
 'view_args': {}}
```

Reply:
That's likely because the MIME type isn't set by the requester. Try setting the MIME type to `application/json` in the request and see what happens.

Reply:
That works
https://github.com/guardicore/monkey/pull/2145/files/c098afaee1c519f20fe8ad592121bf921e613198..d56951f15216434c553b8316302a2f8c618f233a